src/Extension/FluentExtension.php

```diff
@@ -316,6 +321,20 @@ public function augmentDataQueryCreation(SQLSelect $query, DataQuery $dataQuery)
     }
 
     /**
+     * Get the localised table name
+     *
+     * @return string
+     */
+    public function getLocalisedTable()
```

Comment:
How about moving this to a method that takes:
- normal table
- optional locale (for alias)

That way we can use it for the non-base tables also.